src/v/rpc/transport.cc

```diff
@@ -59,52 +64,66 @@ transport::transport(
     }
 }
 
-ss::future<> base_transport::do_connect() {
+ss::future<ss::connected_socket> connect_with_timeout(
+  ss::socket& sock,
+  const seastar::socket_address& address,
+  rpc::clock_type::time_point timeout) {
+    auto f = sock.connect(
+      address,
+      ss::socket_address(sockaddr_in{AF_INET, INADDR_ANY, {0}}),
+      ss::transport::TCP);
+    return ss::with_timeout(timeout, std::move(f))
+      .handle_exception_type([&sock, &address](const ss::timed_out_error& e) {
+          rpclog.info("connection to: {}, timed out", address);
+          sock.shutdown();
+          return ss::make_exception_future<ss::connected_socket>(e);
+      });
+}
+
+ss::future<> base_transport::do_connect(rpc::clock_type::time_point timeout) {
     // hold invariant of having an always valid dispatch gate
     // and make sure we don't have a live connection already
     if (is_valid() || _dispatch_gate.is_closed()) {
-        return ss::make_exception_future<>(std::runtime_error(fmt::format(
+        throw std::runtime_error(fmt::format(
```

Comment:
Why change from an exception future to raw exception?

Reply:
because this method is now a coroutine

Reply:
speaking of, @mmaslankaprv  we should update to use the vlog.h `fmt_with_ctx` to get the line number of th exception